subsys/ipc/ipc_service/backends/Kconfig

```diff
@@ -1,42 +1,32 @@
 # Copyright (c) 2021 Nordic Semiconductor (ASA)
 # SPDX-License-Identifier: Apache-2.0
 
-choice IPC_SERVICE_BACKEND
-	prompt "IPC service backend"
+DT_COMPAT_ZEPHYR_IPC_OPENAMP_STATIC_VRINGS := zephyr,ipc-openamp-static-vrings
+DT_COMPAT_ZEPHYR_IPC_ICMSG := zephyr,ipc-icmsg
+
+config IPC_SERVICE_REG_BACKEND_PRIORITY
+	int "Initialization priority of modules registering IPC backend"
+	default 46
```

Comment:
any chance to have static assert which would verify priority correctness? Something like: https://github.com/zephyrproject-rtos/zephyr/blob/93b0db781eafe9fbff24df6b4d31324c76c6941b/soc/arm/nordic_nrf/nrf53/sync_rtc.c#L310

Reply:
Actually i just moved it, i don't know what was the intended relation. @carlocaione what should we put there?

Reply:
The only relation I can think of is with the MBOX driver that should be probed before the backend. Adding a `BUILD_ASSERT` for that is ok to me.

Reply:
uhmmm, why is it `default 46` then? Default MBOX_INIT_PRIORITY is `Kernel:50`. For backend it would be `Kernel:46` now. Lower initialization priority means earlier initialization. Shouldn't it be > 50 then?
@carlocaione

Reply:
yeah, but that's not really import for the static vrings backend. That backend is initializing the mbox when opening the instance not on init. This change could be useful for other backends that are initializing the mbox during init.

Reply:
so, yeah, change the priority with respect to the mbox and add the assert even though that makes no difference for the current backend.

Reply:
Shall we keep this option at all? I think there is no point in keeping that if any of the backends actually depends on `MBOX_INIT_PRIORITY`. Shall we use `CONFIG_KERNEL_INIT_PRIORITY_DEVICE` for both backends we have now?

Reply:
I think there is no good reason to multiply options here if not needed. @nordic-krch  what do you think?

Reply:
please keep it. The point is that you can have some backend that is depending on some other driver that must be probed before the backend. So you can enforce a probe ordering only by acting on this symbol.